server/graphql/v2/mutation/RootMutations.ts

```diff
@@ -49,6 +49,61 @@ const BanAccountResponse = new GraphQLObjectType({
  * Root mutations
  */
 export default {
+  editAccountFlags: {
+    type: new GraphQLNonNull(Account),
+    description: '[Root only] Edits account flags (deleted, banned, archived, trusted host)',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account to change the flags for',
+      },
+      isDeleted: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is soft deleted',
+      },
+      isBanned: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is banned',
+      },
+      isArchived: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is archived',
+      },
+      isTrustedHost: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is a trusted host',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<Record<string, unknown>> {
+      checkRemoteUserCanRoot(req);
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true, paranoid: false });
+
+      if (args.isDeleted && !account.deletedAt) {
+        await account.destroy();
```

Comment:
:no_entry_sign: We can't just delete the account like that; what if it has attached transactions, hosted collectives, etc? Also, the associated data needs to be deleted too. The `deleteCollective` mutation is supposed to take care of these aspects, let's find a way to re-use the code here.

I suspect this will make the implementation of "restore" more difficult, so I suggest opening an issue to discuss this if you feel that it's too big of a scope.